### DISCUSSION TITLE
[Noob] How to search urls?

### DISCUSSION THREAD

**Discussion by rieje** [BurntSushi/ripgrep#2788] [Q&A] [answered]
Noob question: how to search urls?

    rg 'https://www.youtube.com/watch?v=vqMYnvkgqGd' /tmp/file

doesn't produce but replacing with `grep` does. I understand rg is not supposed to replace grep and also am not suggesting this is a bug. 

Thank you.

**Comment by BurntSushi** [comment, marked as answer]
`grep` uses BREs by default, where things like `?` aren't treated as special meta characters by default. Instead, you have to do `\?`. In contrast, ripgrep is closer to `grep -E` by default, where I imagine you'd get the same result as ripgrep here.

The correct thing to do here, whether it's grep or ripgrep, is to use the `-F/--fixed-literals` flag. That will treat the pattern as literal text completely.
